src/ansys/fluent/core/fluent_connection.py

```diff
@@ -85,6 +92,50 @@ def __call__(self):
         return self._scheme_eval.scheme_eval("(data-valid?)")
 
 
+class FluentConnectionProperties:
+    """Stores Fluent connection properties, including connection IP, port and password;
+    Fluent Cortex working directory, process ID and hostname;
+    and whether Fluent was launched in a docker container.
+
+    Examples
+    --------
+    These properties are also available through the session object and can be accessed as:
+
+    >>> import ansys.fluent.core as pyfluent
+    >>> session = pyfluent.launch_fluent()
+    >>> session.connection_properties.list_properties()
+    ['ip', 'port', 'password', 'cortex_pwd', 'cortex_pid', 'cortex_host', 'inside_container']
+    >>> session.connection_properties.ip
+    '127.0.0.1'
+    """
+
+    def __init__(
+        self,
+        ip: str = None,
+        port: int = None,
+        password: str = None,
+        cortex_pwd: str = None,
+        cortex_pid: int = None,
+        cortex_host: str = None,
+        inside_container: bool = None,
+    ):
+        self.ip = ip
+        self.port = port
+        self.password = password
+        self.cortex_pwd = cortex_pwd
+        self.cortex_pid = cortex_pid
+        self.cortex_host = cortex_host
+        self.inside_container = inside_container
+
+    def list_properties(self) -> list:
+        """List all property names."""
+        return [k for k, _ in vars(self).items()]
+
+    def list_properties_values(self) -> dict:
```

Comment:
the pattern for this simple object seems a little verbose.    this could be even be only a dictionary.

Reply:
True, that is also only a convenience function. I would suggest for now changing it to simply `list_values`, and the other method to just `list`. Going to see what other changes are suggested and then bundle them all up in a single commit

Reply:
Wasn't aware of frozen dataclasses, that is exactly what I wanted, thank you again Mainak

Putting together Patrick's suggestions and making FluentConnectionProperties a frozen dataclass